test/jdk/com/sun/management/HotSpotDiagnosticMXBean/DumpThreads.java

```diff
@@ -103,21 +107,23 @@ public void testJson() throws Exception {
                 mbean.dumpThreads(file.toString(), ThreadDumpFormat.JSON);
                 cat(file);
 
-                assertTrue(count(file, "threadDump") >= 1L);
-                assertTrue(count(file, "time") >= 1L);
-                assertTrue(count(file, "runtimeVersion") >= 1L);
-                assertTrue(count(file, "threadContainers") >= 1L);
-                assertTrue(count(file, "threads") >= 1L);
+                // parse the JSON text
+                String jsonText = Files.readString(file);
+                ThreadDump threadDump = ThreadDump.parse(jsonText);
+
+                // find the thread container that corresponds to the executor
+                String name = Objects.toIdentityString(executor);
```

Comment:
I don't understand the need for `toIdentityString()` rather than just `toString()`

Reply:
The implementation lends on the string returned by Object::toString when not overridden. In this case, either will do so I can change it to toString.